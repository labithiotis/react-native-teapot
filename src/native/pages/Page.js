import React, { Component, PropTypes, cloneElement } from 'react';
import { defaultsDeep } from 'lodash';
import { appStyles } from '../styles';
import { routes } from '../../common/routes';
import { Answers } from 'react-native-fabric';
import { AdMobBanner } from 'react-native-admob';
import { View, StatusBar } from 'react-native';

export default Contents =>
  class Page extends Component {
    static propTypes = {
      navigation: PropTypes.object.isRequired,
      screenProps: PropTypes.object.isRequired,
    };

    static navigationOptions = Contents.navigationOptions;

    constructor(props) {
      super(props);
      this.state = {
        showBanner: true,
      };
    }

    bannerReceiveAd() {
      Answers.logCustom('AdMob Shown');
    }

    bannerPressed() {
      Answers.logCustom('AdMob Pressed');
    }

    bannerError() {
      this.setState({ showBanner: false });
    }

    getRouteOptions() {
      return defaultsDeep(routes[this.props.navigation.state.routeName], {
        screenOptions: {},
        navigationOptions: {},
      });
    }

    render() {
      const { screenProps: { device, config } } = this.props;
      const {
        screenOptions: { adBanner, statusBar },
        navigationOptions: { header },
      } = this.getRouteOptions();
      const styling = { paddingTop: 20 };
      const views = [<Contents {...this.props} />];

      if (this.state.showBanner && adBanner) {
        let addMethod = adBanner === 'top' ? views.unshift : views.push;
        addMethod.call(
          views,
          <AdMobBanner
            style={{ height: 50 }}
            bannerSize={'smartBannerPortrait'}
            adUnitID={
              device.platform === 'ios'
                ? config.adBannerAPI.ios
                : config.adBannerAPI.android
            }
            didFailToReceiveAdWithError={this.bannerError.bind(this)}
            adViewDidReceiveAd={this.bannerReceiveAd.bind(this)}
            adViewWillLeaveApplication={this.bannerPressed.bind(this)}
          />,
        );
      }

      if ((statusBar && statusBar.hidden) || header !== null) {
        styling.paddingTop = 0;
      }

      views.push(
        <StatusBar
          hidden={false}
          animated={true}
          barStyle="light-content"
          showHideTransition="fade"
          {...statusBar}
        />,
      );

      return (
        <View style={[appStyles.container, styling]}>
          {views.map((view, index) => cloneElement(view, { key: index }))}
        </View>
      );
    }
  };
