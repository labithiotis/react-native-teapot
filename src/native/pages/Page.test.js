import React from 'react';
import { View } from 'react-native';
import { shallow } from 'enzyme';

import Page from './Page';

const Child = () => <View />;
const Parent = Page(Child);

describe('Page', () => {
  let component;

  beforeEach(() => {
    const props = global.defaultProps({ test: 'test' });
    component = shallow(<Parent {...props} />);
  });

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should contain a passed in component with props', () => {
    const child = component.find(Child);
    expect(child).toBeTruthy();
    expect(child.prop('test')).toEqual('test');
  });
});
