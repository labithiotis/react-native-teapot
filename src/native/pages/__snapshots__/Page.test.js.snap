// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#000000",
        "flex": 1,
      },
      Object {
        "paddingTop": 0,
      },
    ]
  }
>
  <AdMobBanner
    adUnitID="ca-app-pub-XXX/XXX"
    adViewDidReceiveAd={[Function]}
    adViewWillLeaveApplication={[Function]}
    bannerSize="smartBannerPortrait"
    didFailToReceiveAdWithError={[Function]}
    style={
      Object {
        "height": 50,
      }
    }
  />
  <Child
    navigation={
      Object {
        "navigate": [Function],
        "state": Object {
          "routeName": "home",
        },
      }
    }
    screenProps={
      Object {
        "config": Object {
          "adBannerAPI": Object {
            "android": "ca-app-pub-XXX/XXX",
            "ios": "ca-app-pub-XXX/XXX",
          },
          "animations": Object {
            "delayBetween": 60,
            "enabled": true,
            "maxDelay": 1000,
          },
          "imageFallBack": "https://goo.gl/MrJGPK",
          "imageURL": "https://www.reddit.com/",
          "productPainsURL": "https://productpains.com/",
          "pushNotifications": Object {
            "apiKey": "XXX",
            "senderID": "XXX",
            "url": "",
          },
          "url": "https://www.reddit.com",
        },
        "device": Object {
          "platform": "ios",
        },
        "locale": Object {
          "errors": Object {},
          "pages": Object {},
        },
        "locales": Object {
          "availableLanguages": Array [
            "en",
          ],
          "locales": Object {
            "en": Object {
              "errors": Object {},
              "pages": Object {},
            },
          },
          "selectedLanguage": "en",
        },
        "ui": Object {},
      }
    }
    test="test"
  />
  <StatusBar
    animated={false}
    barStyle="light-content"
    hidden={true}
    showHideTransition="fade"
  />
</View>
`;
